pkg/api/pod/util.go

```diff
@@ -416,6 +416,12 @@ func GetValidationOptionsFromPodSpecAndMeta(podSpec, oldPodSpec *api.PodSpec, po
 		}
 
 		opts.AllowPodLifecycleSleepActionZeroValue = opts.AllowPodLifecycleSleepActionZeroValue || podLifecycleSleepActionZeroValueInUse(podSpec)
+
+		// If old spec has enabled both the feature gates of "InPlacePodVerticalScaling" and "SidecarContainers" and has resize policy set on the restartable init container,
+		// we must allow it
+		if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) && utilfeature.DefaultFeatureGate.Enabled(features.SidecarContainers) {
```

Comment:
@tallclair I am not exactly sure if we need to check feature gates before setting `AllowSidecarResizePolicy`.

Reply:
No, we explicitly do not want to check the feature gates here. The idea is, if the oldpod had a ResizePolicy set on a sidecar, it must have been set on a future version, because that's the only way it's allowed in this version. We don't want to break the pod in that case, so we continue to allow it, even if the features are disabled.

Reply:
Sounds good.